base/utf32.jl

```diff
@@ -1,42 +1,17 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-## UTF-32 in the native byte order, i.e. plain old character arrays ##
-
-immutable UTF32String <: DirectIndexString
-    data::Vector{Char} # includes 32-bit NULL termination after string chars
-
-    function UTF32String(a::Vector{Char})
-        if length(a) < 1 || a[end] != Char(0)
-            throw(ArgumentError("UTF32String data must be NULL-terminated"))
-        end
-        new(a)
-    end
-end
-UTF32String(data::Vector{UInt32}) = UTF32String(reinterpret(Char, data))
-
+# UTF-32 basic functions
 next(s::UTF32String, i::Int) = (s.data[i], i+1)
 endof(s::UTF32String) = length(s.data) - 1
 length(s::UTF32String) = length(s.data) - 1
 
-utf32(x) = convert(UTF32String, x)
-convert(::Type{UTF32String}, c::Char) = UTF32String(Char[c, Char(0)])
-convert(::Type{UTF32String}, s::UTF32String) = s
+reverse(s::UTF32String) = UTF32String(reverse!(copy(s.data), 1, length(s)))
 
-function convert(::Type{UTF32String}, s::AbstractString)
-    a = Array(Char, length(s) + 1)
-    i = 0
-    for c in s
-        a[i += 1] = c
-    end
-    a[end] = Char(0) # NULL terminate
-    UTF32String(a)
-end
+sizeof(s::UTF32String) = sizeof(s.data) - sizeof(Char)
 
 function convert(::Type{UTF32String}, data::AbstractVector{Char})
     len = length(data)
-    d = Array(Char, len + 1)
-    d[end] = Char(0) # NULL terminate
-    UTF32String(copy!(d,1, data,1, len))
+    @inbounds return UTF32String(setindex!(copy!(Vector{Char}(len+1),1,data,1,len),0,len+1))
```

Comment:
Can you find a link to that comment by @vtjnash? Did he specifically say that the existing code on these lines should change at all? You're going to have to take a moving average of all the review you get since everyone's style is a little different, as you've probably noticed by now.